dart_io_mini_samples/example/websockets/using_websockets_server.dart

```diff
@@ -13,17 +13,15 @@ handleMsg(msg) {
 }
 
 main() {
-  runZoned(() {
-    HttpServer.bind('127.0.0.1', 4040).then((server) {
-      server.listen((HttpRequest req) {
-        if (req.uri.path == '/ws') {
-          // Upgrade a HttpRequest to a WebSocket connection.
-          WebSocketTransformer.upgrade(req).then((socket) {
-            socket.listen(handleMsg);
-          });
-        }
-      });
-    });
+  runZoned(() async {
+    var server = await HttpServer.bind('127.0.0.1', 4040);
+    await for (HttpRequest req in server) {
+      if (req.uri.path == '/ws') {
+        // Upgrade a HttpRequest to a WebSocket connection. 
+        var socket = await WebSocketTransformer.upgrade(req);
+        socket.listen(handleMsg);
+      }
+    }
```

Comment:
Would it be possible/practical to make main() async and then use try-catch instead of onError below?